### ISSUE TITLE
📚 `noUndeclaredDependencies` and `useImportType` report issues with output from TypeScript code actions

### ISSUE THREAD

**Issue by chrisgrieser** [biomejs/biome#2027]
### Documentation URL

https://biomejs.dev/linter/rules/use-import-restrictions/

### Description

Using typescript code actions `Add missing imports` results in code that is reported by biome. I am using the default typescript language server settings. 

```ts
// biome reports `noUndeclaredDependencies`
import MyModule from "src/main"; // <- this line is auto-generated by "add missing imports"

// biome does not report`noUndeclaredDependencies` when changed to this:
import MyModule from "../main";
```

```ts
// biome reports `useImportType`
import MyModule from "module"; // <- this line is auto-generated by "add missing imports"
// …
let mod: MyModule;
```

While I think the first can be resolved by changing the typescript configuration to `importModuleSpecifierPreference = relative` ([see here](https://github.com/typescript-language-server/typescript-language-server/blob/master/docs/configuration.md)), I am not aware of any typescript setting to deal with the latter.

### Expectations

That a recommended rule (`useImportType`) reports output of typescript as an issue is at least somewhat irritating, and also quite confusing for less experienced users.

You could also argue that this is somewhat a bug of those rules, but technically, the rules work as expected. Thus, I am not reporting this as a bug, but as a documentation issue. 

Possible solutions to this:

#### A) Change the documentation
1. Mention that these rules oppose the output of typescript code actions with default settings.
2. Explain which typescript settings need to be changed to make the output of the code actions comply with biome's rules.

However, 2. is complicated by the fact that there is (to my knowledge) no typescript setting to make typescript add type imports instead of imports. 

Also, Changing typescript configs is not always straightforward, e.g. for nvim, a few settings can only be changed by using a typescript language server replacement like [typescript tools](https://github.com/pmizio/typescript-tools.nvim).

#### B) Change rule behavior
Alternatively, the rule's behavior should be modified to be less strict. `noUndeclaredDependencies` could be modified to check if an import is simply project-relative, before reporting a violation.

Not sure how to deal with `useImportType` though.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
I have to take some time to write a complete answer. However, as a preliminary answer for `useImportType`: there is an LSP option named `typescript.preferences.preferTypeOnlyAutoImports`. This should partially solve your issue.
